src/io.jl

```diff
@@ -95,23 +102,33 @@ function Base.write(io::IO, d::MRCData; compress=:none, unit_vsize=4096)
     data = parent(d)
     fswap = bswapfromh(h.machst)
     unit_vsize = div(unit_vsize, sizeof(T))
+    if isnothing(buffer)
+        buffer = Vector{T}(undef, unit_vsize)
+    end
+    buffer::Vector{T}
```

Comment:
This way we provide an informative error message:
```suggestion
    elseif !(buffer isa Vector{T})
        throw(ArgumentError("`buffer` must be `nothing` or a `Vector{$T}`, but `$(typeof(buffer))` was provided"))
    end
```